### ISSUE TITLE
When switching grenades, hide animation doesn't play and grenade just disappears visually

### ISSUE THREAD

**Issue by Xottab-DUTY** [Swartz27/xray-16#1124]

**Comment by yohjimane** [comment]
I was hoping this could be fixed by #1359 but sadly not.

At first glance, it looks like some sort of state issue.

I added some debug logs, and got this output when switching grenade types:
```cpp
---To Slot: activating slot [4], Frame[34968]
Yohji debug - start anm_hide at 137845 // same time as state_hidden

-[HUD] as[0] [257]anim_play [anm_hide][34968]
Yohji debug - start state_hidden at 137845 // same time as anm_hide
Yohji debug - start anm_show at 137848 // starts playing 3 ms after anm_hide

-[HUD] as[4] [16]anim_play [anm_show][34969]

-[HUD] as[4] [16]anim_play [anm_idle][35351]

-[HUD] as[4] [16]anim_play [anm_idle][35351]
```

versus unequipping and then reequipping one grenade:
```cpp
Yohji debug - start anm_hide at 151500 // different time vs state_hidden

-[HUD] as[4] [16]anim_play [anm_hide][41905]
Yohji debug - start state_hidden at 152071 // different time vs anm_hide
Yohji debug - start anm_show at 152678 // starts playing 1178ms after anm_hide

-[HUD] as[4] [16]anim_play [anm_show][42482]

-[HUD] as[4] [16]anim_play [anm_idle][42849]

-[HUD] as[4] [16]anim_play [anm_idle][42849]
```

When changing grenade type, the [eHidden case](https://github.com/OpenXRay/xray-16/blob/dev/src/xrGame/Missile.cpp#L269) is hit at the same time that anm_hide is played.

**Comment by yohjimane** [comment]
Ported & refactored fix from Anomaly here, code review pls!

https://github.com/OpenXRay/xray-16/pull/1362/files
